audformat/core/utils.py

```diff
@@ -285,8 +285,49 @@ def duration(
     return (ends - starts).sum()
 
 
+def expand_file_path(
+        index: pd.Index,
+        root: str,
+) -> pd.Index:
+    r"""Expand index with relative file path.
+
+    Args:
+        index: index with relative file path conform to
+            :ref:`table specifications <data-tables:Tables>`
+        root: directory expand the file path with
+
+    Returns:
+        index with absolute file path
+
+    Raises:
+        ValueError: if index is not conform to
+            :ref:`table specifications <data-tables:Tables>`
+
+    Example:
+        >>> index = filewise_index(['f1', 'f2'])
+        >>> index
+        Index(['f1', 'f2'], dtype='object', name='file')
+        >>> expand_file_path(index, '/some/where')  # doctest: +SKIP
+        Index(['/some/where/f1', '/some/where/f2'], dtype='object', name='file')
+
+    """  # noqa: E501
+    if len(index) == 0:
+        return index
+
+    root = audeer.safe_path(root)
+    is_segmented = index_type(index) == define.IndexType.SEGMENTED
+
+    if is_segmented:
+        files = root + os.path.sep + index.levels[0]
+        index = index.set_levels(files, level=0)
```

Comment:
In #120 I do this in a single line:

```python
index.set_levels(root + index.levels[0], level='file')
```
I think your solution is easier to read, but could you test that there is no difference in speed between one line vs. two lines?

Reply:
In both solutions it has to calculate `root + index.levels[0]` first, so I don't see how there could be any measureable difference.

Reply:
But let's simply switch to your solution :)

Reply:
done